static-catalog/README.md

```diff
@@ -1,28 +1,52 @@
-# STAC Static Catalog Spec
+# STAC Catalog Spec
```

Comment:
I want to review the 'naming' a bit. I fully agree we need to shift the naming, but I'm not sure STAC Catalog and STAC API are the best names. I'm definitely cool to merge this though, as it's a ton of great improvements. 

But the 'SpatioTemporal Asset Catalog Catalog Spec' is the expanded version of 'STAC Catalog Spec', which is a bit weird. Also I think this spec should also have the /stac OpenAPI definition in it. Hopefully it's not a huge pain to build from the fragments for a /stac endpoint in this folder. But I'd ideally like to explain what the core 'catalog' is, and then express it in both OpenAPI and JSON Schema versions. And explain the core concept of the catalog here, and explain it can be done with static files or a dynamic server.

Not super set on the exact dividing between folders, but I do think we should introduce 'catalogs' and then from there explain the different implementation options (static (on s3, etc with links) and dynamic (implemented with a server)).

Reply:
Not sure about the name. Do you have a proposal?

The other things seem to be a separate PR, maybe by @hgs-truthe01 ?